api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -22,18 +32,31 @@
 /**
  * This class represents a list of patientIds.
  */
+@Entity
+@Table(name = "cohort")
 @Audited
 public class Cohort extends BaseChangeableOpenmrsData {
 	
 	public static final long serialVersionUID = 0L;
 	
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY, generator = "cohort_id_seq")
+	@GenericGenerator(
+		name = "cohort_id_seq",
+		strategy = "native",
+		parameters = @Parameter(name = "sequence", value = "cohort_cohort_id_seq")
+	)
+	@Column(name = "cohort_id", nullable = false)
 	private Integer cohortId;
 	
+	@Column(name = "name", nullable = false)
 	private String name;
 	
+	@Column(name = "description", nullable = false)
 	private String description;
 	
-	private Collection<CohortMembership> memberships;
+	@OneToMany(mappedBy = "cohort", cascade = CascadeType.ALL, orphanRemoval = true)
+	private Set<CohortMembership> memberships;
```

Comment:
@dilankavishka what is the reason for changing the reference from Collection to Set?

Reply:
You have changed the return type and method signature of some methods. We should avoid that as it could cause problems within modules that use those methods.

Reply:
Refer to this PR: https://github.com/openmrs/openmrs-core/commit/60f3a26f15be46956f16cb1751737b3823476057

Reply:
Let me revert them back again then.

Reply:
Updated